connections.go

```diff
@@ -0,0 +1,245 @@
+package packngo
+
+import (
+	"path"
+)
+
+type ConnectionRedundancy string
+type ConnectionType string
+
+const (
+	connectionBasePath                           = "/connections"
+	virtualCircuitsBasePath                      = "/virtual-circuits"
+	ConnectionShared        ConnectionType       = "shared"
+	ConnectionDedicated     ConnectionType       = "dedicated"
+	ConnectionRedundant     ConnectionRedundancy = "redundant"
+	ConnectionPrimary       ConnectionRedundancy = "primary"
+)
+
+type ConnectionService interface {
+	OrganizationCreate(string, *ConnectionCreateRequest) (*Connection, *Response, error)
+	ProjectCreate(string, *ConnectionCreateRequest) (*Connection, *Response, error)
+	OrganizationList(string, *GetOptions) ([]Connection, *Response, error)
+	ProjectList(string, *GetOptions) ([]Connection, *Response, error)
+	Delete(string) (*Response, error)
+	Get(string, *GetOptions) (*Connection, *Response, error)
+	Events(string, *GetOptions) ([]Event, *Response, error)
+	PortEvents(string, string, *GetOptions) ([]Event, *Response, error)
+	VirtualCircuitEvents(string, *GetOptions) ([]Event, *Response, error)
+	Ports(string, *GetOptions) ([]ConnectionPort, *Response, error)
+	Port(string, string, *GetOptions) (*ConnectionPort, *Response, error)
+	VirtualCircuits(string, string, *GetOptions) ([]ConnectionVirtualCircuit, *Response, error)
+	VirtualCircuit(string, *GetOptions) (*ConnectionVirtualCircuit, *Response, error)
+	DeleteVirtualCircuit(string) (*Response, error)
+}
+
+type ConnectionServiceOp struct {
+	client *Client
+}
+
+type connectionPortsRoot struct {
+	Ports []ConnectionPort `json:"ports"`
+}
+
+type virtualCircuitsRoot struct {
+	VirtualCircuits []ConnectionVirtualCircuit `json:"virtual_circuits"`
+	Meta            meta                       `json:"meta"`
```

Comment:
Somewhat unrelated to this PR, but I wonder why `meta` is not exported.

https://github.com/packethost/packngo/blob/master/packngo.go#L38-L39

I can't find the linter rule or best practice (https://golang.org/doc/effective_go.html or https://github.com/golang/go/wiki/CodeReviewComments) that states this, but IIRC it is something like "exported fields in unexported types are difficult to use".

Reply:
@displague Maybe the meta only makes sense in context of listing API request.